File: My-Homeworks-ExamPreparations-And-Excersizes/CSharp-Advanced/03. Methods/Homework/P04. Appearance count/P04. Appearance count.cs
/*
 *Appearance count
Description

Write a method that counts how many times given number appears in a given array. 
Write a test program to check if the method is working correctly.

Input

On the first line you will receive a number N - the size of the array
On the second line you will receive N numbers separated by spaces - the numbers in the array
On the third line you will receive a number X
Output

Print how many times the number X appears in the array
Constraints

1 <= N <= 1024
Time limit: 0.1s
Memory limit: 16MB
Sample tests

Input	Output
8
28 6 21 6 -7856 73 73 -56
73	2
Submission 
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace P04.Appearance_count
{
    class Program
    {
        static void Main(string[] args)
        {
            int name = int.Parse(Console.ReadLine());
            List<int> nums = ReadInLineArray();
            int elementToCount = int.Parse(Console.ReadLine());

            Console.WriteLine(AppearanceCount(nums, elementToCount));
        }
        
        static List<int> ReadInLineArray()
        {
            string inLine = Console.ReadLine();
            //string inLine = "28 6 21 6 -7856 73 73 -56";

            char[] delimiters = new char[] { ' ', ',' };
            List<int> nums = inLine.Split(delimiters, StringSplitOptions.RemoveEmptyEntries).Select(d => int.Parse(d)).ToList();

            return nums;
        }

        static int AppearanceCount(List<int> nums, int element)
        {
            int counter = 0;

            for (int i = 0; i < nums.Count; i++)
            {
                if (nums[i] == element)
                {
                    counter++;
                }
            }

            return counter;
        }
    }

}

File: My-Homeworks-ExamPreparations-And-Excersizes/JavaScript Fundamentals/08. Functions/Homework/P04. Appearance count/P04. Appearance count.js
/**
 * Created by Admin on 29.6.2016 г..
 */
/**
 * Appearance count

 Description

 Write a method that counts how many times given number appears in a given array.
 Write a test program to check if the method is working correctly.

 Input
 •On the first line you will receive a number N - the size of the array
 •On the second line you will receive N numbers separated by spaces - the numbers in the array
 •On the third line you will receive a number X

 Output
 •Print how many times the number X appears in the array

 Constraints
 •1 <= N <= 1024
 •Time limit: 0.2s
 •Memory limit: 16MB

 Sample tests


 Input

 Output


 8
 28 6 21 6 -7856 73 73 -56
 73

 2

 Submission

 */

// var input = `8
// 28 6 21 6 -7856 73 73 -56
// 73`;

// var input = [ '8', '28 6 21 6 -7856 73 73 -56', '73' ];
// solve(input);

function solve(args) {

    var N = parseInt(args[0]);
    var countedValue = parseInt(args[2]);
    var inputArr = ("" + args[1]);
    inputArr = inputArr.replace(/(\r\n|\n|\r)/g," ");
    inputArr = inputArr.trim(function (s) {
            var sTrim = s.trim();
            var ret = sTrim !== undefined && sTrim !== null && sTrim !== NaN;
            return ret;
        });
    inputArr = inputArr.split(/\s+/g);
    inputArr = inputArr.map(function (item) {
            return parseInt(item, 10);
        });

    // var N = parseInt(inputArr[0]);
    // var countedValue = inputArr[inputArr.length - 1];
    var numbers = new Array();
     for (i = 0; i < N; i = i + 1) {
        numbers[i] = inputArr[i];
    }

    var counter = appearanceCounter(numbers, countedValue);

    console.log(counter);


    function appearanceCounter(arrToSearch, valueToCount) {
        var bestCount = 0;
        for(i = 0; i < arrToSearch.length; i = i +1){
            var isMatch = (arrToSearch[i] === valueToCount);
            if (isMatch){
                bestCount = bestCount + 1;
            }
        }

        return bestCount;
    }

}


/**
 * Copied
 */
// function solve(params) {
//     var arr = (params + '').split('\n');
//     var arrToSearch = arr[1].split(' ');
//     var count = 0;
//     for(var i = 0; i < arrToSearch.length; i+=1){
//         if (+arrToSearch[i] == +arr[2]) {
//             count++;
//         }
//     }
//     console.log(count);
// }